### ISSUE TITLE
scan: include_simulator_logs collects months worth of log data

### ISSUE THREAD

**Issue by lilyball** [fastlane/fastlane#16960]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Now that `include_simulator_logs` finally works, I did a run and my collected logarchive contained about 3.5 months worth of data totaling 500MB. This is extremely unwieldy.

It looks like scan is just trying to collect everything. Instead it should take note of the time prior to launching the tests, and then collect with the `--start` flag to only grab logs from the relevant time frame.

**Comment by lilyball** [comment]
`fastlane env` isn't particularly helpful here. I'm passing `include_simulator_logs: true` in my scan args, and scan is invoking `log collect` on the simulator without providing either `--start` or `--last`, so it collects everything.

**Comment by lilyball** [comment]
Last time I checked this is still an issue. This was reproduced with fastlane 2.153.1, but the release notes for the later releases show nothing relevant, and [the responsible code](https://github.com/fastlane/fastlane/blob/8c9809ed26393626d72402d8c8fd98db918c80ca/fastlane_core/lib/fastlane_core/device_manager.rb#L349) is still just blindly fetching everything.

**Comment by lilyball** [comment]
This is the line of code in question:

```ruby
command = "xcrun simctl spawn #{device.udid} log collect --output #{logarchive_dst.shellescape} 2>/dev/null"
```

Note how it doesn't limit the collection, which means it's going to try and collect *everything*.

**Comment by lyndsey-ferguson** [comment]
This is how I would solve it, by allowing simulator arguments: https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues/286

**Comment by lyndsey-ferguson** [comment]
FYI: I've prioritized this feature and am working on it now.

**Comment by lyndsey-ferguson** [comment]
@lilyball FYI, it looks like the other developer wants to be able to spawn independent activities in the simulator, and you just want to be able to limit the logs collected to the current session. So, this feature, here would be simply note down the start date  of the current session of `scan` and then collect the logs starting from that date. It would seem that that should be the _default_ and an option can be provided to override that if for some reason, someone wants _all_ the logs.

Thoughts?

**Comment by lyndsey-ferguson** [comment]
@lilyball Can you modify your `Gemfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane', :git => "https://github.com/fastlane/fastlane.git", :branch => "add-log-collection-starttime"
```

If there are still problems, please let me know and attach the console output as a text file to this issue (makes it easier for me to review).  If it works, please let me know.

**Comment by lyndsey-ferguson** [comment]
This issue should be closed, as it was already fixed and released as part of [_fastlane_ 2.164.0](https://github.com/fastlane/fastlane/releases/tag/2.164.0) 🚀

**Comment by rogerluan** [comment]
Thanks for your contribution @lyndsey-ferguson 🚀 
If you still experience issues (or new issues arise), please let us know @lilyball 🤗
